### ISSUE TITLE
Better values replace

### ISSUE THREAD

**Issue by shoomyth** [tivac/modular-css#245]
Thanks for this awesome lib!

I'm trying to use this lib's values instead of postcss-simple-vars/advanced/etc.
But as I see replacing of values is very limited in current implementation.

input.css:

```css
@value val: 10px;
@value val2: calc(val - 2px);

.button {
    height: val;
    margin-top: -val;
    padding-bottom: val2;
}
```

Output:

```css
/* src/input.css */
.mc3ba0c2c3_button {
    height: 10px;
    margin-top: -val;
    padding-bottom: 10px
}
```

Expected:

```css
/* src/input.css */
.mc3ba0c2c3_button {
    height: 10px;
    margin-top: -10px;
    padding-bottom: calc(10px - 2px)
}
```

I think this could be improved, but maybe I'm missing something (e.g. specification limitations)?

**Comment by pygy** [comment]
`-val` is a valid CSS identifier, but a weird one... a stand-alone vendor prefix... `calc(0 - val)` could (be made to?) work...

`val2` looks like a bug...

**Comment by shoomyth** [comment]
> -val is a valid CSS identifier, but a weird one... a stand-alone vendor prefix...

Yes, and something like `font-family: -apple-system` is valid as well.
But negative values in css are widely used and imo there is should be a simple solution for them. Using calc() looks really weird for this, and besides requires postcss-calc.

Maybe ES6-style expression interpolation, e.g.:

input.css:
```css
@value apple: google;

.button {
    font-family: -apple-system;
    font-family: -(apple)-system;
}
```

output.css:
```css
.button {
    font-family: -apple-system;
    font-family: -google-system;
}
```

But this is just one of many(?) use-cases. What I want to say in this issue is that values-replace should be more aggressive.

**Comment by tivac** [comment]
I definitely left it pretty cautious on purpose, I'll take a look at the options and try to stroke a better balance.

**Comment by tivac** [comment]
Just a note that I haven't forgotten about this, just been a bit swamped.

**Comment by shoomyth** [comment]
Good to hear. Have you investigated something for this issue? And maybe have some plans for this?

**Comment by tivac** [comment]
This is fixed in `modular-css@3.1.0` which should be on npm shortly.
